datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/nulls.rs

```diff
@@ -193,6 +193,14 @@ pub fn set_nulls_dyn(input: &dyn Array, nulls: Option<NullBuffer>) -> Result<Arr
                 ))
             }
         }
+        DataType::Struct(_) => unsafe {
+            let input = input.as_struct();
+            Arc::new(StructArray::new_unchecked(
```

Comment:
can you please add a safety note here (probably can just reuse the same as above)